### PR TITLE
TF-1414: [Presentation] Create new UI and implement get AllMailBox an…

### PR DESCRIPTION
**Pull request by nqhhdev**
Create new UI and implement get AllMailBox and Subscribe Mailbox

Test cases :
- Hide/Show a mailbox
- Hide/Show parent of list mailbox
- Show default mailbox: if `isSubscribe == false` => display text button `Show` else Nothing

Demo:
- `WEB`

https://user-images.githubusercontent.com/99852347/218439450-6215920d-7bbf-4dbe-bee5-ea5b84d99da7.mov
- `Mobile`
[Mobile.webm](https://user-images.githubusercontent.com/99852347/218440061-bb86832c-83d2-4efe-be4f-bc7c83627ee2.webm)

### CODE REVIEW

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildLoadingView() {
+    return Obx(() => controller.viewState.value.fold(
+      (failure) => const SizedBox.shrink(),
+      (success) => success is LoadingState
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:72]: I did not see the loading indicator, need to wait with a blank page, after longtime, mailboxes are showed. please check it

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+          const SizedBox(height: 8),
+          Obx(() => controller.personalMailboxIsNotEmpty
+              ? _buildMailboxCategory(context, MailboxCategories.personalMailboxes, controller.personalRootNode)
+              : const SizedBox.shrink()),
+
+        ])
+    );
+  }
+
+  Widget _buildHeaderMailboxCategory(BuildContext context, MailboxCategories categories) {
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:109]: this is duplicated?

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+              maxLines: 1,
+              overflow: TextOverflow.ellipsis,
+              style: const TextStyle(
+                  fontSize: 17,
+                  color: Colors.black,
+                  fontWeight: FontWeight.bold))),
+        ]));
+  }
+
+  Widget _buildMailboxCategory(BuildContext context, MailboxCategories categories, MailboxNode mailboxNode) {
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:135]: this is duplicated?

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        AnimatedContainer(
+            duration: const Duration(milliseconds: 400),
+            child: categories.getExpandMode(controller.mailboxCategoriesExpandMode.value) == ExpandMode.EXPAND
+                ? _buildBodyMailboxCategory(context, categories, mailboxNode)
+                : const Offstage())
+      ]),
+    );
+  }
+
+  Widget _buildBodyMailboxCategory(
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:155]: idem

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+            left: _responsiveUtils.isDesktop(context) ? 0 : 12),
+        child: TreeView(
+            key: Key('${categories.keyValue}_mailbox_list'),
+            children: _buildListChildTileWidget(
+                context,
+                mailboxNode,
+                lastNode: lastNode)));
+  }
+
+  List<Widget> _buildListChildTileWidget(
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:174]: idem?

File: lib/features/manage_account/presentation/mailbox_visibility/widgets/mailbox_visibility_folder_tile_builder.dart [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:model/mailbox/expand_mode.dart';
+import 'package:tmail_ui_user/features/mailbox/domain/extensions/presentation_mailbox_extension.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_actions.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_node.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations.dart';
+
+typedef OnExpandFolderActionClick = void Function(MailboxNode);
+typedef OnSubscribeMailboxActionClick = void Function(MailboxNode);
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/mailbox_visibility/widgets/mailbox_visibility_folder_tile_builder.dart:11]: Please do the cross-review :) @dab246 define a place to keep all the action for mailbox. We have a lot of these

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart (hunk 2 above)
Comment (nqhhdev) [reply lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:109]: No duplicate, it's different for this screen

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart (hunk 1 above)
Comment (nqhhdev) [reply lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:72]: Demo this case 

`Web`

https://user-images.githubusercontent.com/99852347/218651323-bb70f677-0383-4720-8b3c-9a4102ffd307.mov

`Mobile`

[Mobile.webm](https://user-images.githubusercontent.com/99852347/218651489-da49bca8-be9f-48c4-995e-ff417ae9c5a8.webm)

File: lib/features/mailbox/presentation/mailbox_controller.dart [hunk 7]
```diff
@@ -198,7 +201,7 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
   void _registerObxStreamListener() {
     ever(mailboxDashBoardController.accountId, (accountId) {
       if (accountId is AccountId) {
-        getAllMailboxAction(mailboxDashBoardController.sessionCurrent!, accountId);
+        _getAllMailboxAction(mailboxDashBoardController.sessionCurrent!, accountId);
```
Comment (hoangdat) [inline lib/features/mailbox/presentation/mailbox_controller.dart:204]: why not directly invoke `getAllMailbox(session, accountId);`?

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+    MailboxNode parentNode,
+    {MailboxNode? lastNode}
+  ) {
+    return parentNode.childrenItems
+        ?.map((mailboxNode) => mailboxNode.hasChildren()
+          ? TreeViewChild(
+              context,
+              key: const Key('children_tree_mailbox_child'),
+              isExpanded: mailboxNode.expandMode == ExpandMode.EXPAND,
+              parent: Obx(() => controller.defaultMailboxIsNotEmpty ? (MailBoxVisibilityFolderTileBuilder(context, _imagePaths, mailboxNode, lastNode: lastNode)
```
Comment (dab246) [inline lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:202]: why only check `defaultMailboxIsNotEmpty`? What about `personalMailbox` and `teamMailbox`?

File: lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+                ..addOnExpandFolderActionClick((mailboxNode) => controller.toggleMailboxFolder(mailboxNode))
+                ..addOnSubscribeMailboxActionClick((mailboxNode) => controller.subscribeMailbox(mailboxNode)))
+                .build(context) : const SizedBox.shrink()) ,
+              children: _buildListChildTileWidget(context, mailboxNode)
+          ).build()
+          : Obx(() => controller.defaultMailboxIsNotEmpty ? (MailBoxVisibilityFolderTileBuilder(context, _imagePaths, mailboxNode, lastNode: lastNode)
```
Comment (dab246) [inline lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_view.dart:208]: same

**Review by dab246 — APPROVED** [review]

**Pull request merged**